### PR TITLE
Add scooze_id to Card

### LINKED ISSUE #192: Add scooze_id to Cards
### Description
As part of the convert Deck<>DeckModel issue #70 , we discovered that it would not be possible to lookup cards in the database unless the Card class had some notion of scooze_id. We are adding it here. And to future proof for Tournaments needing to look up decks by id, we are adding it to Deck as well.

### PR DESCRIPTION
**Pull request by iambroadband**
Fixes #192

### CODE REVIEW

File: src/scooze/card.py [hunk 1]
```diff
@@ -43,6 +44,7 @@ class Card(HashableObject):
     use to sort a decklist.
 
     Attributes:
+        scooze_id: A unique identifier for a document in a scooze database.
```
Comment (iambroadband) [inline src/scooze/card.py:47]: remove this

File: src/scooze/card.py [hunk 2]
```diff
@@ -87,14 +91,23 @@ def __str__(self):
 
     @classmethod
     def from_json(cls, data: dict | str) -> Self:
+        """
```
Comment (iambroadband) [inline src/scooze/card.py:94]: remove this

File: src/scooze/card.py [hunk 3]
```diff
@@ -87,14 +91,23 @@ def __str__(self):
...
         if isinstance(data, dict):
             return cls(**data)
         elif isinstance(data, str):
             return cls(**json.loads(data))
 
     @classmethod
-    def from_model(cls, model: CardModel) -> Self:
-        return cls(**model.model_dump())
+    def from_model(cls, model: CardModelOut) -> Self:
+        """
```
Comment (iambroadband) [inline src/scooze/card.py:105]: remove this

File: src/scooze/card.py [hunk 4]
```diff
@@ -87,14 +91,23 @@ def __str__(self):
...
+        TODO: docstring - mention that this updates the scoozeid?
+        """
+
+        card = cls(**model.model_dump())
```
Comment (iambroadband) [inline src/scooze/card.py:109]: change this back to 1 liner

**Comment by iambroadband** [comment]
https://stackoverflow.com/a/26027865

re: mutating the Card that's passed to the API

File: src/scooze/models/card.py [hunk 5]
```diff
@@ -614,7 +614,7 @@ class CardModelIn(FullCardModel):
 
 
 class CardModelOut(FullCardModel):
-    id: ObjectIdT = Field(
+    scooze_id: ObjectIdT = Field(
```
Comment (fjork3) [inline src/scooze/models/card.py:617]: just to make sure: is this going to be a breaking change for data loaded into DB with an earlier version?

File: tests/api/test_card_api.py [hunk 6]
```diff
@@ -9,61 +9,60 @@
 @patch("scooze.database.card.get_card_by_property")
 def test_get_base_card(mock_get: MagicMock, recall_base, asyncio_runner):
     model = CardModelOut.model_validate(recall_base.__dict__)
-    model.id = ObjectId()
+    # model.scooze_id = recall_base.scooze_id
```
Comment (fjork3) [inline tests/api/test_card_api.py:12]: can remove this line, and similar commented ones below

File: src/scooze/models/card.py (hunk 5 above)
Comment (iambroadband) [reply src/scooze/models/card.py:617]: Yes.

File: tests/api/test_card_api.py (hunk 6 above)
Comment (iambroadband) [reply tests/api/test_card_api.py:12]: ok

File: src/scooze/api/card.py [hunk 7]
```diff
@@ -89,8 +89,10 @@ async def add_card(card: CardT) -> ObjectId:
 
     card_model = CardModelIn.model_validate(card.__dict__)
     model = await db.add_card(card=card_model)
+
     if model is not None:
-        return model.id
+        # TODO: update card scooze id here?
```
Comment (iambroadband) [inline src/scooze/api/card.py:94]: @fjork3

**Review by fjork3 — APPROVED** [review]

**Review by JacobGinsparg — APPROVED** [review]

**Pull request merged**